File: 프로그래머스1단계/제일작은수제거하기.cpp
#include <string>
#include <vector>

using namespace std;

//변수 잘 둬방...

vector<int> solution(vector<int> arr) {
    vector<int> answer;
    int min = 100000;
    int minIndex;
    
    for(int i = 0; i < arr.size(); i++){
        if(min > arr[i]){
            min = arr[i];
            minIndex = i;
        }
    }
    
    for(int i = 0; i < arr.size(); i++){
        if(i == minIndex){
            continue;
        }
        
        answer.push_back(arr[i]);
    }
    
     if(arr.size()  <= 1){
        answer.push_back(-1);
        return answer;
    }
    else
        return answer;
}

File: 프로그래머스1단계/제일작은수제거하기.java
import java.util.*;

class 제일작은수제거하기 {
    public int[] solution(int[] arr) {
        int[] answer = new int[arr.length-1];
        int min = arr[0];
        int cnt = 0;
        if(arr.length <= 1){
            return new int[] {-1};
        }
        
        for(int i=1; i < arr.length; i++){
            if(min > arr[i]){
                min = arr[i];
            }
        }
        
        for(int i=0; i < arr.length; i++){
            if(min == arr[i]){
                continue;
            }
            else{
                answer[cnt++] = arr[i];
            }
        }
        
        return answer;
    }
}